readthedocs/docsitalia/signals.py

```diff
@@ -87,3 +90,20 @@ def on_publisher_project_delete(sender, instance, **kwargs):  # noqa
     if projects_pks:
         instance.projects.filter(pk__in=projects_pks).delete()
         clear_es_index.delay(projects=projects_pks)
+
+
+@receiver(post_save, sender=Project)
+def index_documents_project_save(instance, *args, **kwargs):
+    from readthedocs.search.documents import PageDocument
+
+    html_obj_ids = HTMLFile.objects.filter(project=instance).values_list('id', flat=True)
```

Comment:
we only need to update this if the project used to be the default one

Reply:
we need to update this for `default_version` and previous `default_version`.
So I can get them in `pre_save` signal or by caching old values in Models `__init__` (don't want to change RTD models).